### PR TITLE
CPLP-1227 bulk-upload own company shared idp users

### PR DESCRIPTION
**Pull request by ntruchsess**
implements https://jira.catena-x.net/browse/CPLP-1227

### CODE REVIEW

File: src/framework/Framework.IO/CsvParser.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            {
+                throw new ControllerArgumentException($"value for {itemName} type string expected", documentParameterName ?? DefaultDocumentParameterName);
+            }
+            yield return items.Current;
+        }
+    }
+
+    public static async ValueTask<(int Processed, int Lines, IEnumerable<(int Line, Exception Error)> Errors)> ProcessCsvAsync<TLineType>(
+        Stream stream,
+        Action<string> validateFirstLine,
```
Comment (Phil91) [inline src/framework/Framework.IO/CsvParser.cs:86]: I'm thinking about the implementation of the first line validation, we could also hand over the HeaderColumns and validate those, than we don't need to execute an action from outside. But I'm not sure about that. The positive effect of passing the action from outside is that we could validate all kind of stuff (even service calls).

**Review by Phil91 — CHANGES_REQUESTED** [review]
Please provide unit tests, in my opinion espacially the csv parser is perfect to test with specific csv files for each case we could add into the test.

**Comment by ntruchsess** [comment]
I did refactor the existing file-upload that is used in identityprovider user-migration to use the new framework implementation, so no csv-upload with individually implementation of the algorithm remains. Please check.

**Review by Phil91 — COMMENTED** [review]
changes in identityprovider looks fine, as everything else. we have 6 code smells, we should look into.
About the code duplication, I'm not sure why's that. Can you tell what sonars problem is?

**Comment by ntruchsess** [comment]
All smells are fixed. To make sonar happy I had to apply small refactoring to the usercreation to reduce the complexity.
I did refactor all usecases that create companyUsers to make use of the UserCreationService-implementation. In addition to that I fixed the open 'TODO' in userdeletion not covering own-ipd or mixed idp scenarios.
Can you please re-check?

**Comment by ntruchsess** [comment]
regarding the duplication: I don't think sonar has a problem. There is some duplication (the stream implementations) that technically cannot be avoided as one has to implement the whole interface.

**Review by Phil91 — COMMENTED** [review]
Code changes look fine to me. We should at least add UnitTests for the Upload

**Comment by ntruchsess** [comment]
@Phil91 guess test-coverage is now sufficient ;-). Can you please do a final re-review?

**Review by Phil91 — CHANGES_REQUESTED** [review]
Review done. Looks good so far. Tested it - works fine.

Locally one Test is running into an error - TestAsyncCopyToSuccess in AsyncEnumerableStringStreamTest, is it only on my machine or do you get the error as well?

If its on my machine I'm going to approve it

**Review by Phil91 — APPROVED** [review]
Looks really good! 👍

**Pull request merged**